### PR TITLE
[galactic] spatial: Fix behavior after USB reattachment

### PR DESCRIPTION
**Pull request by mintar**
The Phidged Spatial never recovered after detaching and reattaching to the USB port. This commit fixes that.

The cause of the failure was the following:

* After reattachment, the device time stamp restarts from 0. This caused our driver to throw the following error messages:

       [ WARN]: Time went backwards [...]! Not publishing message.

* However, the data interval is also reset to the default of 256 ms. If the parameter data_interval_ms is set to something else, this caused the arriving data to always be outside the acceptable window for synchronization. Therefore, synchronization never happened, and the driver never resumed publishing messages.

This commit fixes the bug by setting the appropriate data interval after each reattachment and forcing a resynchronization immediately.

### CODE REVIEW

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
Approved both this one and the one for Foxy.

@mintar Once you've merged these in, I'll suggest doing a release, at least for Rolling.  That should fix the currently broken patch in the release repository.  Thanks!

**Comment by mintar** [comment]
@clalancette wrote:

> @mintar Once you've merged these in, I'll suggest doing a release, at least for Rolling. That should fix the currently broken patch in the release repository. Thanks!

Yes, that was exactly my plan. I've been planning to do releases for foxy, galactic and rolling.

I've merged both PRs, did `catkin_generate_changelog` and `catkin_prepare_release`, but I'm stuck at the `bloom-release` step because I don't have write access to the https://github.com/ros2-gbp/phidgets_drivers-release repository (I've only done ROS1 releases so far). Could you give me access, please?

**Comment by clalancette** [comment]
@mintar I've opened up https://github.com/ros2-gbp/ros2-gbp-github-org/pull/11 , which should restore your access once it is merged and deployed.  I'll let you know when that happens.

**Comment by clalancette** [comment]
All right, you should have permissions now.  Let me know if you are still having trouble.

**Comment by nuclearsandwich** [comment]
> All right, you should have permissions now. Let me know if you are still having trouble.

In point of fact, you have likely received an invitation to join the ros2-gbp organization which you'll need to accept in order to use the ros2-gbp release repository.

**Comment by mintar** [comment]
Thanks! I've released phidgets_drivers into noetic, foxy, galactic and rolling now.

**Pull request merged**
